### PR TITLE
jmh setup

### PR DESCRIPTION
**Pull request by ivansenic**
* before we merge I will make my decision if we should specify the benchmark mode on the test class, or by default always use the default which is throughput.. I am not sure in general what are we more interested in?
*  I updated Gradle to `5.2.1`
* read the `PERFTEST.md` first 
* please check carefully my updates in the non-test classes

### CODE REVIEW

File: inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java [hunk 1]
```diff
@@ -211,12 +211,10 @@ public static InspectitContext createFromCurrent(CommonTagsManager commonTagsMan
...
     @Override
     public void makeActive() {
-
-        boolean anyDownPropagatedDataOverwritten = dataOverwrites.keySet().stream()
-                .anyMatch(propagation::isPropagatedDownWithinJVM);
+        boolean anyDownPropagatedDataOverwritten = anyDownPropagated();
 
         //only copy if any down-propagating value has been written
-        if (anyDownPropagatedDataOverwritten) {
+        if (anyDownPropagated()) {
```
Comment (JonasKunz) [inline inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: you are not using the boolean variable here which you declared above

File: inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java [hunk 2]
```diff
@@ -238,6 +236,15 @@ public void makeActive() {
         }
     }
 
+    private boolean anyDownPropagated() {
```
Comment (JonasKunz) [inline inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: rename maybe to "anyDownPropagatedDataOverridden" ?

File: inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java [hunk 3]
```diff
@@ -384,13 +391,11 @@ private void readOverridesFromCurrentTagContext() {
 
     private Stream<Map.Entry<String, Object>> getDataAsStream() {
         return Stream.concat(
-                postEntryPhaseDownPropagatedData.entrySet().stream()
-                        .filter(e -> !dataOverwrites.containsKey(e.getKey())),
-                dataOverwrites.entrySet().stream()
-                        .filter(e -> e.getValue() != null));
+                postEntryPhaseDownPropagatedData.entrySet().stream().filter(e -> !dataOverwrites.containsKey(e.getKey())),
```
Comment (JonasKunz) [inline inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: did you intentionally remove the line breaks here?

**Comment by JonasKunz** [comment]
I somehow get a "Error reading benchmark list" after excecuting the JMH task for the first time successfully.
Even a gradle clean does not help.

File: inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java (hunk 1 above)
Comment (ivansenic) [reply inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: :+1: great catch..

File: inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java (hunk 2 above)
Comment (ivansenic) [reply inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: done

File: inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java (hunk 3 above)
Comment (ivansenic) [reply inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: hmm seems so, maybe I tried to optimize here and then reverted it.. Look easier to read for `Stream.concat`, should I revert?

Comment (JonasKunz) [reply inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/InspectitContext.java]: Matter of taste, I'm fine with both

**Pull request merged**